src/org/opendatakit/briefcase/ui/reused/SourceSelectPanelForm.java

```diff
@@ -30,17 +39,37 @@ public SourceSelectPanelForm() {
   }
 
   private void triggerConfigure() {
-    AggregateServerConnectionDialog dialog = AggregateServerConnectionDialog.empty(__ -> true);
-    dialog.onConnect(this::trigureConfiguredSource);
-    dialog.getForm().setVisible(true);
+    String source = (String) sourceComboBox.getSelectedItem();
```

Comment:
This can be null. Wrap it inside an Optional

Reply:
I think it's time for an Enum. This could work:

```java
interface Source<T> {
  Source<AggregateServerConnectionConfiguration> AGGREGATE = callback -> {
    AggregateServerConnectionDialog dialog = AggregateServerConnectionDialog.empty(__ -> true);
    dialog.onConnect(callback);
    dialog.getForm().setVisible(true);
  };
  Source<Path> CUSTOM_DIR = callback -> {
    WrappedFileChooser fc = new WrappedFileChooser(this,
        new ODKCollectFileChooser(this));

    int retVal = fc.showDialog();

    if (retVal == JFileChooser.APPROVE_OPTION) {
      Path path = Paths.get(fc.getSelectedFile().getAbsolutePath());
      callback.accept(path);
    }
  };

  void onSelect(Consumer<T> callback);
}

private void triggerConfigure() {
  Optional
      .ofNullable((Source<?>) sourceComboBox.getSelectedItem())
      .ifPresent(source -> {
        if (source.equals(AGGREGATE))
          AGGREGATE.onSelect(this::triggerAggregateConnection);
        if (source.equals(CUSTOM_DIR))
          CUSTOM_DIR.onSelect(this::triggerCustomDirSource);
      });
}
```

Reply:
Should I place this in the same class or create a new interface?

Reply:
New file :)

Reply:
If I declare it in a new file, I cannot access the container required for making the file chooser. Is there a workaround for this?

Reply:
If you change `onSelect` to receive a `Container` and the `Consumer<T>`, you could pass it along with the callback to a `Source`